lib/l4p/udp_stream.c

```diff
@@ -276,6 +276,74 @@ tle_udp_stream_open(struct tle_ctx *ctx,
 	return &s->s;
 }
 
+	struct tle_stream *
+tle_udp_stream_establish(struct tle_ctx *ctx,
+		const struct tle_udp_stream_param *prm)
+{
+	struct tle_udp_stream *s;
+
+	if (ctx == NULL || prm == NULL || check_stream_prm(ctx, prm) != 0) {
+		rte_errno = EINVAL;
+		return NULL;
+	}
+
+	s = (struct tle_udp_stream *)get_stream(ctx);
+	if (s == NULL)	{
+		rte_errno = ENFILE;
+		return NULL;
+
+		/* some TX still pending for that stream. */
+	} else if (UDP_STREAM_TX_PENDING(s)) {
+		put_stream(ctx, &s->s, 0);
+		rte_errno = EAGAIN;
+		return NULL;
+	}
+
+	/* copy input parameters. */
+	s->prm = *prm;
+
+	/* setup L3 src and dst addresses. */
+	if (prm->local_addr.ss_family == AF_INET) {
+		s->s.type = TLE_V4;
+		s->s.ipv4.addr.dst = ((const struct sockaddr_in *)&prm->local_addr)->sin_addr.s_addr;
+		s->s.ipv4.mask.dst = INADDR_NONE;
+		s->s.ipv4.addr.src = ((const struct sockaddr_in *)&prm->remote_addr)->sin_addr.s_addr;
+		s->s.ipv4.mask.src = INADDR_NONE;
+	} else if (prm->local_addr.ss_family == AF_INET6) {
+		s->s.type = TLE_V6;
+		const struct in6_addr *pm;
+		pm = &tle_ipv6_none;
+		//fill_ipv6_am((const struct sockaddr_in6 *)&prm->local_addr,
+		//	&s->s->ipv6.addr.dst, &s->s->ipv6.mask.dst);
+		memcpy(&s->s.ipv6.addr.dst, &((const struct sockaddr_in6 *)&prm->local_addr)->sin6_addr, sizeof(s->s.ipv6.addr.dst));
+		memcpy(&s->s.ipv6.mask.dst, pm, sizeof(s->s.ipv6.mask.dst));
+		//fill_ipv6_am((const struct sockaddr_in6 *)&prm->remote_addr,
+		//	&s->s->ipv6.addr.src, &s->s->ipv6.mask.src);
+		memcpy(&s->s.ipv6.addr.src, &((const struct sockaddr_in6 *)&prm->remote_addr)->sin6_addr, sizeof(s->s.ipv6.addr.src));
+		memcpy(&s->s.ipv6.mask.src, pm, sizeof(s->s.ipv6.mask.src));
+	}
+
+	/* setup L4 port info */
+	s->s.port.src = ((const struct sockaddr_in *)&prm->remote_addr)->sin_port;
+	s->s.pmsk.src = UINT16_MAX;
+	s->s.port.dst = ((const struct sockaddr_in *)&prm->local_addr)->sin_port;
```

Comment:
Nit: Above you've used the ordering dst,src but here you've switched it up.

Reply:
most of it was ordered as it was pulled from other parts of code to make it work.  should there be some standard ordering...probably

Reply:
Gotcha...you know me, haha...that OCD kicks in.

Reply:
well this is fixed but for other reasons.